pkg/storage/snapshot/snapshot.go

```diff
@@ -823,9 +824,9 @@ func (ctrl *VMSnapshotController) updateVolumeSnapshotStatuses(vm *kubevirtv1.Vi
 
 	vmCopy := vm.DeepCopy()
 	var statuses []kubevirtv1.VolumeSnapshotStatus
-	for i, volume := range vmCopy.Spec.Template.Spec.Volumes {
+	for _, volume := range storageutils.GetVolumes(vmCopy, storageutils.FLAG_INCLUDE_ALL) {
 		log.Log.V(3).Infof("Update volume snapshot status for volume [%s]", volume.Name)
-		status := ctrl.getVolumeSnapshotStatus(vmCopy, &vmCopy.Spec.Template.Spec.Volumes[i])
+		status := ctrl.getVolumeSnapshotStatus(vmCopy, &volume)
```

Comment:
Be careful with this. I am fairly certain we want the pointer to the indexed version instead of a pointer to the volume. IIRC the volume value is the same address each iteration. So setting the pointer will mean every single one is the same if you have more than one result.

Hmm maybe not, I just tried this in the playground and it worked fine https://go.dev/play/p/WFHo3Nfc5I5 I think this used to be an issue, but maybe not remembering it correctly.